eolearn/mask/cloud_mask.py

```diff
@@ -41,6 +41,159 @@ def predict_proba(self, X: np.ndarray) -> np.ndarray:  # noqa: N803
+            A value of `0` skips this post-processing step. Default value mimics the default for
+            s2cloudless: `4`.
+        :param dilation_size: Size of the dilation post-processing step. A value of `0` or `None` skips this
+            post-processing step. Default value mimics the default for s2cloudless: `2`.
+        """
+        self.data_feature = self.parse_feature(data_feature)
+        self.valid_data_feature = self.parse_feature(valid_data_feature)
+
+        self.data_indices = None
+        if data_indices is None:
+            s2_l1c_bands = [band.name for band in DataCollection.SENTINEL2_L1C.bands]
+            s2_l1c_model_bands = ["B01", "B02", "B04", "B05", "B08", "B8A", "B09", "B10", "B11", "B12"]
+            self.data_indices = [s2_l1c_bands.index(band) for band in s2_l1c_model_bands]
+
+        self.output_mask_feature = self.parse_feature(output_mask_feature)
+        self.output_proba_feature = None
+        if output_proba_feature is not None:
+            self.output_proba_feature = self.parse_feature(output_proba_feature)
+
+        self.threshold = threshold
+
+        self.avg_kernel = None
+        if average_over is not None and average_over > 0:
+            self.avg_kernel = disk(average_over) / np.sum(disk(average_over))
+
+        self.dil_kernel = None
+        if dilation_size is not None and dilation_size > 0:
+            self.dil_kernel = disk(dilation_size).astype(np.uint8)
+
+        self._classifier: ClassifierType | Booster | None = None
+
+    @staticmethod
+    def _parse_resolution_arg(resolution: None | float | tuple[float, float]) -> tuple[float, float] | None:
+        """Parses initialization resolution argument"""
+        if isinstance(resolution, (int, float)):
+            resolution = resolution, resolution
+
+        return resolution
+
+    @property
+    def classifier(self) -> ClassifierType | Booster:
+        """An instance of a custom-provided cloud classifier. Loaded only the first time it is required."""
+        if self._classifier is None:
+            path = os.path.join(self.MODELS_FOLDER, self.CLASSIFIER_NAME)
+            self._classifier = Booster(model_file=path)
+
+        return self._classifier
+
+    @staticmethod
+    def _run_prediction(classifier: ClassifierType | Booster, features: np.ndarray) -> np.ndarray:
+        """Uses classifier object on given data"""
+        is_booster = isinstance(classifier, Booster)
+
+        predict_method = classifier.predict if is_booster else classifier.predict_proba
+        prediction: np.ndarray = execute_with_mp_lock(predict_method, features)
+
+        return prediction if is_booster else prediction[..., 1]
+
+    def _average(self, data: np.ndarray) -> np.ndarray:
+        return cv2.filter2D(data.astype(np.float64), -1, self.avg_kernel, borderType=cv2.BORDER_REFLECT)
+
+    def _dilate(self, data: np.ndarray) -> np.ndarray:
+        return (cv2.dilate(data.astype(np.uint8), self.dil_kernel) > 0).astype(np.uint8)
```

Comment:
do you think we could move these definitions into `_average_all` and `_dilate_all` or would that be equally messy?

Reply:
will be removed in the next MR